cpp/src/arrow/meson.build

```diff
@@ -696,8 +711,8 @@ pkg.generate(
 subdir('testing')
 
 subdir('array')
-subdir('c')
 subdir('compute')
+subdir('c')
```

Comment:
FWIW this is a minor deviation from the CMake configuration. In the CMake configuration, the `bridge_test` links to the `arrow_compute_XXX` libs, but those libs have not yet been declared in the configuration because they are located in the compute/CMakeLists.txt file, which is included after c/CMakeLists.txt

Without knowing how CMake works, I'm not sure if that's a problem or not, but the Meson configuration definitely complains about `arrow_compute_dep does not exist` if you try to maintain the same configuration inclusion order as CMake